modules/core/src/main/scala/iota/internal/toolbelts.scala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2016-2017 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package iota
+package internal
+
+import cats.Id
+import cats.Foldable
+import cats.syntax.foldable._
+
+import scala.annotation.tailrec
+import scala.collection.immutable.Map
+import scala.reflect.api.Universe
+import scala.reflect.runtime.{ universe => runtimeUniverse }
+import scala.reflect.macros.blackbox.Context
+
+private[internal] object IotaMacroToolbelt {
+  def apply[C <: Context](c: C): IotaMacroToolbelt[c.type] =
+    new IotaMacroToolbelt[c.type](c)
+
+  final class Cache {
+    @volatile var underlying: Map[Any, Any] = Map.empty
+  }
+
+  final val typeListCache: Cache = new Cache()
+}
+
+private[internal] class IotaMacroToolbelt[C <: Context](val c: C)
+    extends IotaCommonToolbelt
+{
+  override type Uu = c.universe.type
+  override val u: Uu = c.universe
+
+  import u._
+
+  lazy val showAborts =
+    !c.inferImplicitValue(typeOf[debug.optionTypes.ShowAborts], true).isEmpty
+
+  lazy val showCache =
+    !c.inferImplicitValue(typeOf[debug.optionTypes.ShowCache], true).isEmpty
+
+  lazy val showTrees =
+    !c.inferImplicitValue(typeOf[debug.optionTypes.ShowTrees], true).isEmpty
+
+  def foldAbort[F[_]: Foldable, T](
+    either: Either[F[String], Tree],
+    isImplicit: Boolean = false
+  ): c.Expr[T] =
+    either fold (
+      errors => {
```

Comment:
I would've gone with:

```
either fold (errors =>
)
```

I don't know if this is ScalaFMT or just preference.